### ISSUE TITLE
Eratic mouse behaviour

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#218]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/218) by Tierry Anthony (Bitbucket: [Tissiwau](https://bitbucket.org/Tissiwau), ).**

----------------------------------------

I updated my compatibility source on 20th November and now have predictable but uncontrollable mouse behaviour. Using the mouse for either motion or for viewing, the view will suddenly jump to either directly overhead, or down to ground level. The 'jump' point is always at the same place and appears to be around a view angle of 45 degrees. More than this I cannot offer at present.

I am currently compiling under Kubuntu 14.04, with Intel i7 Haswell processor and Nvidia GTX 760 Graphics.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

In events_listener.cpp, at ln 188
try changing

```
#!c++

if (s_MouseFreeLookWaitCenter)
```

to

```
#!c++

if (s_MouseFreeLookWaitCenter && scX == 0 && scY == 0)
```

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Also try (@ ln 188)


```
#!c++

			bool outsideBounds = ((abs(scX) > (drW >> 3)) || (abs(scY) > (drH >> 3)));
			if (s_MouseFreeLookWaitCenter && !outsideBounds)
```
and remove the existing outsideBounds line (@ ln 208), replacing if (outsideBounds) with

```
#!c++

if (outsideBounds && !s_MouseFreeLookWaitCenter)
```


This version may be more reliable cross-platform.

**Comment by ryzom-pipeline** [comment]
**Original comment by Tierry Anthony (Bitbucket: [Tissiwau](https://bitbucket.org/Tissiwau), ).**

----------------------------------------

Your first suggestion, of returning line 188 to if (s_MouseFreeLookWaitCenter && scX == 0 && scY == 0), made very little difference but the second option of adding the !outsideBounds check to s_MouseFreeLookWaitCenter at line 188 seems to resolve the problem.

Thank you for your rapid response. I am still very unfamilar with most of the code so it takes me for ever to narrow down a problem.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Ok, thanks. Let me know in a few days if this permanently solves the issue; then I'll patch it in like that.

**Comment by ryzom-pipeline** [comment]
**Original comment by Tierry Anthony (Bitbucket: [Tissiwau](https://bitbucket.org/Tissiwau), ).**

----------------------------------------

Just to confirm that there appears to be no adverse effects from the suggested modifications, after three days of significant use, so I believe that it is safe to permanently patch it into the code.

**Comment by ryzom-pipeline** [comment]
**Original comment by Geyos (Bitbucket: [Geyos](https://bitbucket.org/Geyos), GitHub: [Geyos](https://github.com/Geyos)).**

----------------------------------------

Can confirm, i'm using the fix since three days on debian sid, no side effects using this fix :)

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Fix #218 erratic mouse behaviour
